### DISCUSSION TITLE
Data extraction for MMF (Magnetomotive Force)

### DISCUSSION THREAD

**Discussion by bcaldeirar764** [Eomys/pyleecan#666] [Q&A]
Hello, I'm using the Pyleecan software to generate a .fem file for an electric machine, but I'm having difficulties extracting data related to the unitary magnetomotive force (also known as the stator phase winding function) curve.

Could someone guide me on where to start and point me in the right direction, please? I'm depending on this data for my master's project, as I've tried obtaining it through another software without success as well. 

 I appreciate any assistance, and thank you in advance!

**Comment by BonneelP** [comment]
Hello,

Thank you for your interest in pyleecan. 
If I understood correctly, there is a button on the winding step in the GUI: 
![image](https://github.com/Eomys/pyleecan/assets/20533034/1827dc37-4769-4b13-95c9-86d303d3bad5)
![image](https://github.com/Eomys/pyleecan/assets/20533034/2ce50e84-1676-4508-8c0c-4ce925badfd8)

If this is what you are looking for the corresponding method is machine.stator.plot_mmf_unit that calls machine.stator.comp_mmf_unit (stator is a LamSlotWind object that you can create with the GUI).

Best regards,
Pierre

**Comment by bcaldeirar764** [comment]
Hello, Pierre

First of all, I would like to thank you for the promptness and agility with
which you responded, and forgive me if I misunderstood your previous
explanation.

In the manual for using Pyleecan, the definition of the winding function
appears as shown in the figure (attached screenshot).

Although the winding function I defined also has zero intervals, I cannot
generate all the points for the whole angular variation/position. Even when
I generate these points using the interface I only get the ones with
maximum or minimum amplitude but not the ones when the winding function is
zero.

Is there any way to get all the points on this curve so that I can
reconstruct it in Matlab code and use it in my work?

Thank you so much for the help.

Best regards,

Bárbara

Em seg., 4 de dez. de 2023 12:22, Pierre Bonneel ***@***.***>
escreveu:
